context/app/static/js/shared-styles/accordions/StyledDetails/StyledDetails.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { PropsWithChildren, useRef, ElementRef, AnimationEvent, MouseEvent, useCallback } from 'react';
+import Box from '@mui/material/Box';
+
+type StyledDetailsProps = PropsWithChildren<{
+  summary: React.ReactNode;
+  summaryBackgroundColor?: string;
+}>;
+
+function StyledDetails({ summary, children, summaryBackgroundColor = 'white' }: StyledDetailsProps) {
```

Comment:
Did some customization of the MUI `Accordion` components not fit this design? This is great, but using MUI shields us from maintaining the complexity and provides somewhat consistent accessibility (but this is using native html elements so it could be better!).

Reply:
That's a valid concern! My reasoning for implementing a fully custom component for this was that I'd essentially need to override every single style that the MUI accordion comes with in order to match the design, but after seeing how tricky reproducing its strengths with semantic HTML was, I'm inclined to believe that creating a `DetailAccordion` variant of some sort would be a more maintainable approach.

This was still a really good exercise for learning how to style semantic HTML components/work with attributes, but in hindsight, the maintainability concerns were a major oversight on my part.